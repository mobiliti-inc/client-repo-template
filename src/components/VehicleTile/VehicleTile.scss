@import "../../../styles/_variables.scss";

.vehicle-tile-button {
	display: block;
	float: left;
	overflow: hidden;
	@media (min-width: 500px) {
		height: 410px;
	}
	@media (min-width: 600px) {
		height: 310px;
	}
	@media (min-width: 650px) {
		height: 330px;
	}
	@media (min-width: 700px) {
		height: 365px;
	}
	@media (min-width: 800px) {
		height: 380px;
	}
	@media (min-width: 900px) {
		height: 400px;
	}
	@media (min-width: $tablet-width) {
		height: 330px;
	}
	@media (min-width: $tablet-width + 100) {
		height: 350px;
	}
	@media (min-width: $desktop-width) {
		height: 350px;
	}
}

.vehicle-image-container {
	margin: 0;
	display: block;
	overflow: hidden;
}

.thumbnail {
	width: 100%;
	display: block;
	margin: 0 auto;
}

.vehicle-tile-button {
	padding: 0;
	cursor: pointer;
	box-shadow: none;
	border: none;
	background: none;
	transition: box-shadow .2s linear,
		height .2s linear,
		margin-bottom .2s linear,
		transform .2s ease,
		background .1s ease;
	&:focus {
		outline: none;
	}
	&:hover {
		background: #fff;
		box-shadow: 0 10px 20px 0 rgba(0,0,0,0.19);
		transform: scale(1.04);
		z-index: 99;
		border: none;
		.show-on-hover {
			max-height: 60px;
			opacity: 1;
			.divider {
				height: 4px;
			}
		}		
	}
}

.vehicle-tile {
	display: flex;
	flex-direction: column;
	padding: 0;
	flex: 1;
	height: 390px;
	z-index: 1;
	width: 100%;
	small {
		font-family: $hanken-sans;
		font-size: 14px;
		color: #7B8DA8;
		text-align: left;
		font-weight: 500;
		margin-bottom: 2px;
	}
}

.vehicle-tile-text {
	// padding: 20px 15px;
	text-align: left;
	height: 120px;
	display: flex;
	flex-direction: column;
	justify-content: center;
	.vehicle-tile-middle-align {
		flex: 1;
		flex-grow: 0;
		padding: 0 20px;
	}
}

.divider {
	height: 0;
	background: $blue;
	width: 100%;
	transition: height .2s ease-in-out;
}

.callout-link {
	display: flex;
}

.thetile {
	padding: 40px;
}

.vehicle-info {
	display: flex;
	height: 60px;
}

.name-info {
	display: flex;
	flex-direction: column;
	flex: 3;
}
.title {
	font-family: $headline-font;
	color: $darker-blue;
	font-size: 20px;
	font-weight: 700;
	flex: 1;
	margin: 0 0 2px 0;
	flex-grow: 0;
	text-align: left;
}
.vehicle-tile-button.ie11 {
	.title {
		min-height: 20px;
	}
}
.title.alt {
	color: $blue;
	text-align: right;
}

.pricing-info {
	display: flex;
	flex-direction: column;
	flex: 1;
}
.subtitle {
	font-family: $hanken-sans;
	color: $grey-blue;
	font-size: 12px;
	flex: 1;
	margin: 0;
	text-align: left;
	min-height: 20px;
}
.subtitle.alt {
	color: $blue;
	text-align: right;
}

.vehicle-image {
	background-size: contain;
	min-height: 340px;
	width: auto;
	background-position: 50% 50%;
	background-repeat: no-repeat;
	@media (min-width: 600px) {
		min-height: 0;
		background-size: cover;
		height: 180px;
		width: auto;
	}
	@media (min-width: 650px) {
		height: 220px;
	}
	@media (min-width: 700px) {
		height: 220px;
	}
	@media (min-width: 700px) {
		height: 260px;
	}
	@media (min-width: 900px) {
		height: 300px;
	}
	@media (min-width: $tablet-width) {
		height: 220px;
	}
}

.carat {
	float: right;
}

.link-text {
	flex: 1;
	font-size: 16px;
	font-weight: 100;
	padding: 10px 0 10px 2px;
}
.right {
	text-align: right;
}

.vehicle-image-container {
	position: relative;
	.available {
		display: none;
	}
	.out-on-subscription,
	.coming-soon {
		position: absolute;
		bottom: 20px;
		left: 20px;
		background: rgba(255, 255, 255, 0.7);
		border-radius: 4px;
		font-family: $hanken-sans;
		font-weight: 600;
		font-size: 14px;
		z-index: 9;
		color: #1D3155;
		padding: 8px 15px;
	}
}