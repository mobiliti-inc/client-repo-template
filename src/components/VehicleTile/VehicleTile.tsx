/* eslint-disable camelcase */ // API returns lots of snake_case data
import React from "react";
import { isUndefined } from "util";
import { IMAGES_BASE_URL } from "../../../api";
import { VehicleModel } from "../../models";
import styles from "./VehicleTile.scss";
import StatusMessages, { statusClasses } from "../../constants/vehicleStatusCodes";

type VehicleTileProps = {
	vehicle: any,
	onClick: (...args: any[]) => any
};

class VehicleTile extends React.PureComponent<VehicleTileProps, {}> {
	render() {
		if (isUndefined(this.props.vehicle)) return null;
		const isIE11 = !!window.MSInputMethodContext && !!document.documentMode;
		const {
			vehicle_id,
			vehicle_title,
			subscription_price,
			image_url,
			dealer_name,
			vehicle_status
		} = this.props.vehicle;
		const status = StatusMessages[vehicle_status];
		const vehicleInfoArray = vehicle_title.split(" ");
		vehicleInfoArray.pop();
		const year = vehicleInfoArray.shift();
		const title = vehicleInfoArray.join(" ");
		return (
			<button
				styleName={`vehicle-tile-button ${isIE11 ? "ie11" : ""}`}
				onClick={() => this.props.onClick(vehicle_id)}
			>
				<div styleName="vehicle-tile" id={vehicle_id}>
					<div styleName="vehicle-image-container">
						<div styleName={statusClasses[vehicle_status]}>{status}</div>
						{isIE11 ? (
							<img styleName="thumbnail" src={image_url} alt={`${title}`} />
						) : (
								<div styleName="vehicle-image" style={{ zIndex: -1, backgroundImage: `url(${image_url}` }} />
							)}
					</div>
					<div styleName="vehicle-tile-text">
						<div styleName="vehicle-tile-middle-align">
							<small>{year}</small>
							<div styleName="vehicle-info">
								<div styleName="name-info">
									<h2 styleName="title">{title}</h2>
									<p styleName="subtitle">{dealer_name}</p>
								</div>
								{vehicle_status === 1 && (
									<div styleName="pricing-info">
										<h2 styleName="title alt">${subscription_price}</h2>
										<p styleName="subtitle alt">Starting&nbsp;at</p>
									</div>
								)}
							</div>
						</div>
					</div>
				</div>
			</button>
		);
	}
}

export default CSSModules(VehicleTile, styles, { allowMultiple: true });
