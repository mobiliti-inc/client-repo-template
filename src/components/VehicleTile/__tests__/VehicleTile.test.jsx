import VehicleTile from '../VehicleTile';
import sampleVehicless from '../../../actions/sampleData/sampleVehicles';

describe('VehicleTile', () => {
	it('renders', () => {
		const category = sampleVehicless[0];
		const subject = shallow(<VehicleTile
			onClick={() => { }}
			vehicle={category}
		/>);
		matchSnapshot(subject);
	});
});
