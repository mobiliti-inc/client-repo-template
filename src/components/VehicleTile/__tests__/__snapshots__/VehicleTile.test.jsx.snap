// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleTile renders 1`] = `
<button
  className="vehicle-tile-button"
  onClick={[Function]}
>
  <div
    className="vehicle-tile"
    id={4}
  >
    <div
      className="vehicle-image-container"
    >
      <div
        className="available"
      >
        Available
      </div>
      <div
        className="vehicle-image"
        style={
          Object {
            "backgroundImage": "url(undefined",
            "zIndex": -1,
          }
        }
      />
    </div>
    <div
      className="vehicle-tile-text"
    >
      <div
        className="vehicle-tile-middle-align"
      >
        <small>
          2018
        </small>
        <div
          className="vehicle-info"
        >
          <div
            className="name-info"
          >
            <h2
              className="title"
            >
              Chevy Tahoe
            </h2>
            <p
              className="subtitle"
            >
              Divvy Motors
            </p>
          </div>
          <div
            className="pricing-info"
          >
            <h2
              className="title alt"
            >
              $
            </h2>
            <p
              className="subtitle alt"
            >
              StartingÂ at
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</button>
`;
